DevOps Insights Quality gate fails

IDRA module appears to be throwing an error.  The NPM module for IDRA was updated in the past 12 days and could be a factor.  A workaround is to manually configure TEST stage and update the Quality Gate job to uncheck "Stop running this stage if this job fails".

npm ERR! Linux 4.4.0-137-generic
npm ERR! argv "/home/pipeline/.nvm/versions/node/v6.9.1/bin/node" "/home/pipeline/.nvm/versions/node/v6.9.1/bin/npm" "run" "idra"
npm ERR! node v6.9.1
npm ERR! npm  v3.10.8
npm ERR! code ELIFECYCLE
npm ERR! logistics-wizard-erp@1.0.0 idra: `istanbul cover --report lcov --report json-summary _mocha -- -R xunit -O output=./test/results/mocha.xml 'test/**/*.js' && idra --publishtestresult --filelocation=./coverage/coverage-summary.json --type=code && idra --publishtestresult --filelocation=./test/results/mocha.xml --type=unittest`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the logistics-wizard-erp@1.0.0 idra script 'istanbul cover --report lcov --report json-summary _mocha -- -R xunit -O output=./test/results/mocha.xml 'test/**/*.js' && idra --publishtestresult --filelocation=./coverage/coverage-summary.json --type=code && idra --publishtestresult --filelocation=./test/results/mocha.xml --type=unittest'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the logistics-wizard-erp package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     istanbul cover --report lcov --report json-summary _mocha -- -R xunit -O output=./test/results/mocha.xml 'test/**/*.js' && idra --publishtestresult --filelocation=./coverage/coverage-summary.json --type=code && idra --publishtestresult --filelocation=./test/results/mocha.xml --type=unittest

--- a/.bluemix/pipeline-TEST.sh
+++ b/.bluemix/pipeline-TEST.sh
@@ -5,8 +5,11 @@ curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.31.2/install.sh | b
 nvm install 6.9.1
 npm install
 if [ -z ${COVERALLS_REPO_TOKEN} ]; then
-  npm run idra
   echo No Coveralls token specified, skipping coveralls.io upload
+
+  npm run idra:coverage
+  npm run idra:publish-coverage
+  npm run idra:publish-unittest
 else
   npm run coverage
 fi

--- a/.bluemix/pipeline-TEST.sh
+++ b/.bluemix/pipeline-TEST.sh
@@ -7,8 +7,13 @@ npm install
 if [ -z ${COVERALLS_REPO_TOKEN} ]; then
   echo No Coveralls token specified, skipping coveralls.io upload
 
+  echo Running DevOps Insights coverage and unit tests
   npm run idra:coverage
+
+  echo Publishing coverage results to DevOps Insights
   npm run idra:publish-coverage
+
+  echo Publishing unit tests to DevOps Insights
   npm run idra:publish-unittest
 else
   npm run coverage

--- a/.gitignore
+++ b/.gitignore
@@ -14,13 +14,10 @@
 .idea
 .project
 .strong-pm
-coverage
 node_modules
 npm-debug.log
 client/vendor
 in-memory-database.json
-coverage
 .coveralls.yml
 out
 server/datasources.local.json
-test/results

--- a/coverage/.gitignore
+++ b/coverage/.gitignore
@@ -1,0 +1,4 @@
+# Ignore everything in this directory
+*
+# Except this file
+!.gitignore

--- a/package.json
+++ b/package.json
@@ -12,7 +12,10 @@
     "coverage": "istanbul cover _mocha test/*.js --report lcovonly -- -R spec --timeout 300000 && cat ./coverage/lcov.info | ./node_modules/coveralls/bin/coveralls.js && rm -rf ./coverage/",
     "localcoverage": "istanbul cover _mocha test/*.js --report html -- -R spec --timeout 300000",
     "exportapi": "slc loopback:export-api-def -o spec.yaml",
-    "idra": "istanbul cover --report lcov --report json-summary _mocha -- -R xunit -O output=./test/results/mocha.xml 'test/**/*.js' && idra --publishtestresult --filelocation=./coverage/coverage-summary.json --type=code && idra --publishtestresult --filelocation=./test/results/mocha.xml --type=unittest"
+    "idra:coverage": "istanbul cover --report lcov --report json-summary _mocha -- -R xunit -O output=./test/results/mocha.xml 'test/**/*.js'",
+    "idra:publish-coverage": "idra --publishtestresult --filelocation=./coverage/coverage-summary.json --type=code",
+    "idra:publish-unittest": "idra --publishtestresult --filelocation=./test/results/mocha.xml --type=unittest"
+
   },
   "dependencies": {
     "async": "2.1.4",

--- a/test/results/.gitignore
+++ b/test/results/.gitignore
@@ -1,0 +1,4 @@
+# Ignore everything in this directory
+*
+# Except this file
+!.gitignore

